jsplot/src/main/java/tech/tablesaw/plotly/Utils.java

```diff
@@ -13,18 +14,22 @@ public static String quote(String string) {
     return "'" + string + "'";
   }
 
-  private static String escape(String s) {
-    return s.replaceAll("\\'", "\\\\'");
+  /**
+   * Escapes string for Javascript, assuming but without surrounding it with doublequotes (") and
+   * saves to output to the given StringBuilder.
+   */
+  private static void escape(String s, StringBuilder sb) {
+    BufferRecyclers.quoteAsJsonText(s, sb);
   }
 
   /** @return a Javascript array of strings (escaped if needed) */
   public static String dataAsString(Object[] data) {
     StringBuilder builder = new StringBuilder("[");
     for (int i = 0; i < data.length; i++) {
       Object o = data[i];
-      builder.append("'");
-      builder.append(escape(String.valueOf(o)));
-      builder.append("'");
+      builder.append("\"");
```

Comment:
I'd probably use Jackson to convert the whole array instead of just the value. But this seems like an improvement in any case

Reply:
Yes, if you think about it 90% of the output template is a JSON. We could just marshal an object there with Jackson...

Reply:
FYI, here is a branch where I'm trying to use Jackson instead of PebbleTemplate: https://github.com/emilianbold/tablesaw/commits/emi/jackson

It doesn't look too bad. JS code is indented better. I dislike that it uses quote for member names too (being JSON, not plain JS) but it's manageable.

A lot of work though to finish all those classes...

Reply:
Cool. You don't actually have to change that much though. Jackson can turn a `Map<String, Object>` into JSON correctly

Reply:
> Jackson can turn a Map<String, Object> into JSON correctly

Heh, I actually didn't notice that. I thought using a (bean) object might come handy for hierarchies. For example `HoverLabel.font` used `Font`. But maybe we just even more Maps as values?

Reply:
I think it also handles a mix of beans and maps just fine. I don't know the jsplot code that well so whatever you think makes the most sense. Feel free to merge this whenever. I don't mean to hold it up

Reply:
OK, I'll merge this PR and we can talk about Jackson when I do a PR about that.